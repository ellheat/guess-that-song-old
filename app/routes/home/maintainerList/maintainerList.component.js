import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';

import messages from './maintainersList.messages';
import { Container, Title, List, ListItem } from './maintainerList.styles';

export class MaintainerList extends PureComponent {
  static propTypes = {
    items: PropTypes.object.isRequired,
  };

  render() {
    return (
      <Container>
        <Title>
          <FormattedMessage {...messages.title} />
        </Title>

        <List>
          {this.props.items.toArray().map((item, key) => (
            <ListItem key={key}>
              <a href={`mailto:${item.get('email')}`}>
                {item.get('firstName')} {item.get('lastName')}
              </a>
            </ListItem>
          ))}
        </List>
      </Container>
    );
  }
}
