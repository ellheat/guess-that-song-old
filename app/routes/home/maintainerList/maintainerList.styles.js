import styled from 'styled-components';

export const Container = styled.div`
 display: flex;
 flex-direction: column;
 align-items: center;
 justify-content: center;
 text-align: center;
`;

export const Title = styled.h3`
  margin-bottom: 8px;
`;

export const List = styled.ul`
  list-style: none;
  padding: 0;
  margin: 0;
`;

export const ListItem = styled.li`
  margin: 0;
`;
