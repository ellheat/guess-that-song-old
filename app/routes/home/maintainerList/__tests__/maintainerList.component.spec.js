import React from 'react';
import { shallow } from 'enzyme';
import { fromJS } from 'immutable';

import { MaintainerList } from '../maintainerList.component';


describe('MaintainerList: Component', () => {
  const defaultProps = {
    items: fromJS([{ name: 'name-1' }, { name: 'name-2' }, { name: 'name-3' }]),
  };

  const component = (props) => (
    <MaintainerList {...defaultProps} {...props} />
  );

  it('should render MaintainerList root', () => {
    const wrapper = shallow(component({}));
    global.expect(wrapper).toMatchSnapshot();
  });
});
