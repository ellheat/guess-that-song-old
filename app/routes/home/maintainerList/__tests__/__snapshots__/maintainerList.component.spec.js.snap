// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MaintainerList: Component should render MaintainerList root 1`] = `
<styled.div>
  <styled.h3>
    <FormattedMessage
      defaultMessage="Maintainers"
      id="home.maintainerList.title"
      values={Object {}}
    />
  </styled.h3>
  <styled.ul>
    <styled.li
      key="0"
    >
      <a
        href="mailto:undefined"
      >
         
      </a>
    </styled.li>
    <styled.li
      key="1"
    >
      <a
        href="mailto:undefined"
      >
         
      </a>
    </styled.li>
    <styled.li
      key="2"
    >
      <a
        href="mailto:undefined"
      >
         
      </a>
    </styled.li>
  </styled.ul>
</styled.div>
`;
